import "common.proto";
import "type.proto";

package fesql.dbms;
option cc_generic_services = true;
option java_package = "com._4paradigm.fesql.dbms";
option java_outer_classname = "DBMS";

message AddGroupRequest {
    optional string name = 1;
}

message AddGroupResponse {
    optional common.Status status = 1;
}

message AddDatabaseRequest {
    optional string name = 1;
}

message AddDatabaseResponse {
    optional common.Status status = 1;
}

message EnterDatabaseRequest {
    optional string name = 1;
}

message EnterDatabaseResponse {
    optional common.Status status = 1;
}

message AddTableRequest {
    optional type.TableDef table = 1;
}

message AddTableResponse {
    optional common.Status status = 1;
}

message GetSchemaRequest {
    optional string name = 1;
}

message GetSchemaResponse {
    optional common.Status status = 1;
    optional type.TableDef table = 2;
}

message GetItemsRequest {
    optional string name = 1;
}

message GetItemsResponse {
    optional common.Status status = 1;
    repeated string items = 2;
}

service DBMSServer {
    rpc AddGroup (AddGroupRequest) returns (AddGroupResponse);
    rpc AddDatabase (AddDatabaseRequest) returns (AddDatabaseResponse);
    rpc EnterDatabase (EnterDatabaseRequest) returns (EnterDatabaseResponse);
    rpc AddTable (AddTableRequest) returns (AddTableResponse);
    rpc GetSchema (GetSchemaRequest) returns (GetSchemaResponse);
    rpc GetDatabases (GetItemsRequest) returns (GetItemsResponse);
    rpc GetTables (GetItemsRequest) returns (GetItemsResponse);
}
